File: 3386-button-with-longest-push-time/3386-button-with-longest-push-time.cpp
class Solution {
public:
    int buttonWithLongestTime(vector<vector<int>>& events) {
        map<int, int> ans;
        int last = 0;
        for (auto &t : events) {
            int id = t[0];
            int tt = t[1];
            ans[id] = max(ans[id], tt - last);
            last = tt;
        }
        int mx = 0, idx = 0;
        for (auto &t : ans) {
            if (t.second > mx) {
                mx = t.second;
                idx = t.first;
            }
        }
        return idx;
    }
};

File: 3386-button-with-longest-push-time/3386-button-with-longest-push-time.java
class Solution {
    public int buttonWithLongestTime(int[][] events) {
        int max = -1;
        int arg = -1;
        int b = 0;
        for(int[] e : events){
            int t = e[1] - b;
            if(t > max || t == max && e[0] < arg){
                max = t;
                arg = e[0];
            }
            b = e[1];
        }
        return arg;
    }
}